src/react/components/pages/editorPage/editorPage.test.tsx

```diff
@@ -0,0 +1,151 @@
+import IProjectActions, * as projectActions from "../../../../redux/actions/projectActions";
+import { AssetService } from "../../../../services/assetService";
+import createReduxStore from "../../../../redux/store/store";
+import MockFactory from "../../../../common/mockFactory";
+
+jest.mock("../../../../services/projectService");
+import ProjectService from "../../../../services/projectService";
+import EditorSideBar from "./editorSideBar";
+
+describe("Editor Page Component", () => {
+    let projectServiceMock: jest.Mocked<typeof ProjectService> = null;
+
+    function createCompoent(store, props: IEditorPageProps): ReactWrapper {
+        return mount(
+            <Provider store={store}>
+                <Router>
+                    <EditorPage {...props} />
+                </Router>
+            </Provider>,
+        );
+    }
+
+    beforeEach(() => {
+        projectServiceMock = ProjectService as jest.Mocked<typeof ProjectService>;
+    });
+
+    it("Sets project state from redux store", () => {
+        const testProject = MockFactory.createTestProject("TestProject");
+        const store = createStore(testProject, true);
+        const props = createProps(testProject.id);
+        const loadProjectSpy = jest.spyOn(props.actions, "loadProject");
+
+        const wrapper = createCompoent(store, props);
+        const editorPage = wrapper.find(EditorPage).childAt(0);
+
+        expect(loadProjectSpy).not.toBeCalled();
+        expect(editorPage.prop("project")).toEqual(testProject);
+    });
+
+    it("Loads project assets when state changes", () => {
+        const testProject = MockFactory.createTestProject("TestProject");
+        const store = createStore(testProject, true);
+        const props = createProps(testProject.id);
+
+        const wrapper = createCompoent(store, props);
+        const editorPage = wrapper.find(EditorPage).childAt(0);
+
+        wrapper.setProps({
+            project: testProject,
+        });
+
+        setImmediate(() => {
+            expect(editorPage.prop("project")).toEqual(testProject);
+            expect(editorPage.state("assets")).toEqual([]);
+        });
+    });
+
+    fit("Raises onAssetSelected handler when an asset is selected from the sidebar", async () => {
+        
+        AssetProviderFactory.register("testProvider", () => MockFactory.createAssetProvider());
+        const testProject = MockFactory.createTestProject("TestProject");
+
+        const store = createStore(testProject, true);
+        const props = createProps(testProject.id);
+
+        const wrapper = createCompoent(store, props);
+        const editorPage = wrapper.find(EditorPage).childAt(0);
+
+        const editorSideBar = wrapper.find(EditorSideBar);
+        const onAssetSelectedSpy = jest.spyOn(editorSideBar.props(), "onAssetSelected");
+
+        // THOUGHT PROCESS: probs have to create jest.fn() => something for onAssetSelected
```

Comment:
The sidebar component exposes a public `selectAsset` method where you can tell it to select a given asset... this should then raise the event handler that is bound in the editor page... I will make sure this is working as expected.